core/model.py

```diff
@@ -0,0 +1,230 @@
+from datetime import datetime
+from enum import Enum
+from typing import TypeVar, Union, get_args
+import re
+from bson.objectid import ObjectId
+from flask_restx import fields
+from .database import get_database
+
+_prefilled_fields = ['_id', 'created_at', 'updated_at']
+_filter_out_load_object_key = ['_database', '_fields']
+_filter_out_store_object_key = _filter_out_load_object_key + ['_id']
+
+T = TypeVar('T')
+
+class CustomField(fields.Raw):
+    __schema_type__ = 'some-type'
+    __schema_format__ = 'some-format'
+    __schema_example__ = {}
+    def __init__(self, type: str, format: str, *args, **kwargs):
+        self.__schema_type__ = type
+        self.__schema_format__ = format
+        return super().__init__(*args, **kwargs)
+
+class Model():
+    _id: ObjectId
+    created_at: datetime
+    updated_at: datetime
+
+    def __init__(self, data: dict=dict()):
+        database = get_database()
+        self._database = database()
+        self._fields = self._get_fields()
+        self._set_value_from_data(data)
+
+    def _set_value_from_data(self, data: dict):
+        for key in self._fields:
+            setattr(self, key, data.get(key))
+
+    def _get_data_load_object(self, data: dict) -> dict:
+        data_load_object = {}
+        if not data:
+            return data_load_object
+        for key in data:
+            if key in _filter_out_load_object_key or key not in data or key not in self._fields:
+                continue
+
+            value = data[key]
+            if key in _prefilled_fields: # is _id
+                data_load_object[key] = value
+            elif self._is_reference_field(key): # is one2many reference
+                elements = []
+                reference_model = self._get_reference_model_of_field(key)
+                for element in value:
+                    elements.append(reference_model.get(element))
+                data_load_object[key] = elements
+            elif self.__annotations__[key].__base__ is Model: # is many2one reference
+                data_load_object[key] = self.__annotations__[key].get(value)
+            elif self.__annotations__[key].__base__ is Enum: # is enum
+                data_load_object[key] = self.__annotations__[key](value)
+            else: # is general type
+                data_load_object[key] = value
+        return data_load_object
+
+    def _get_data_store_object(self) -> dict:
+        data_store_object = {}
+        for key in self.__dict__.keys():
```

Comment:
You can iterate through keys and values with `.items()`

Reply:
fixed. 0d30a47323334a25a1984fdcf6e4ff261288b8f7